vmware-base/src/com/cloud/hypervisor/vmware/util/VmwareClient.java

```diff
@@ -152,13 +152,26 @@ public void connect(String url, String userName, String password) throws Excepti
         @SuppressWarnings("unchecked")
         Map<String, List<String>> headers = (Map<String, List<String>>)((BindingProvider)vimPort).getResponseContext().get(MessageContext.HTTP_RESPONSE_HEADERS);
         List<String> cookies = headers.get("Set-cookie");
+
+        vimPort.login(serviceContent.getSessionManager(), userName, password, null);
+
+        if (cookies == null) {
+            @SuppressWarnings("unchecked")
```

Comment:
to suppress the warnings for the unchecked cast from Object to Map<String,List<String>> on extracting the http response headers from the response context. This is also set on vmware sample programs.